### DISCUSSION TITLE
It's not possible to override the Accept-Language request header

### DISCUSSION THREAD

**Discussion by StefanoRaffaelli** [cefsharp/CefSharp#4665] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

120.1.110.0

### Operating System

Windows 11

### Architecture

x64

### .Net Version

.NET 4.8

### Implementation

WinForms

### Reproduction Steps

I have replicated the problem using the last version of CefSharp.MinimalExample that uses the 120.1.110.0 CEF version

I have modified BrowserForm.cs adding these pieces of code:

* Into the BrowserForm constructor I have added 
`browser.RequestHandler = new CustomRequestHandler();`

* Here the CustomRequestHandler implementation

```
private class CustomRequestHandler : RequestHandler
{
    protected override IResourceRequestHandler GetResourceRequestHandler(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame,
        IRequest request, bool isNavigation, bool isDownload, string requestInitiator, ref bool disableDefaultHandling)
    {
        return new CustomResourceRequestHandler();
    }
}

```

* Here the CustomResourceRequestHandler implementation

```
private class CustomResourceRequestHandler : ResourceRequestHandler
{
    protected override CefReturnValue OnBeforeResourceLoad(IWebBrowser chromiumWebBrowser, IBrowser browser,
        IFrame frame, IRequest request,
        IRequestCallback callback)
    {
        var headers = request.Headers;
        headers.Add("Accept-Language", "it-IT");
        headers.Add("MyCustomHeader", "CustomHeader");
        request.Headers = headers;

        return CefReturnValue.Continue;
    }
}
```

as you can see I have overridden the Accept-Language header and I have also added a new custom header.

When I debug the code I see that `request.Headers` contains the 'it-IT' value into the Accept-Language header , but when I see the request of the URL on devtools the Accept-Language contains only the default value (en-US,en;q=0.9)

Note that no other changes were made to the original project (CefSharp.MinimalExample)

### Expected behavior

I expect the request to the URL to have the Accept-Language header modified (in my case must contains the 'it-IT' value) and the custom header added

### Actual behavior

The request starts with the default Accept-Language header but the CustomHeader is correctly added to the request. Below, the request headers from the dev-tools form
 
![image](https://github.com/cefsharp/CefSharp/assets/4578769/16153833-2616-4450-8315-fc7034405ad7)

### Regression?

In my opinion, it's not a regression because in my application I have used the CEF 107.1.90.0 version and I have the same behavior

### Does this problem also occur in the CEF Sample Application

Not Tested

**Comment by amaitland** [comment]
Converting to discussion as this isn't a bug.

> I expect the request to the URL to have the Accept-Language header modified (in my case must contains the 'it-IT' value) and the custom header added

It's not possible to set `Accept-Language` header currently.

https://github.com/chromiumembedded/cef/blob/e4acacee18d66d9bfe65a15072307d619117bd32/libcef/browser/net_service/resource_request_handler_wrapper.cc#L801

There's an open issue to improve this https://github.com/chromiumembedded/cef/issues/2881

Use either `CefSettings` or `RequestContextSettings` to set the langauge.
